### PR TITLE
Support es2017

### PR DESCRIPTION
**Pull request by digitalsadhu**
Builds polyfill to Es2017, replaces rollup with esbuild (because building back to es2017 was just easier/less dependencies and the actual bundle output was slightly smaller than with Rollup). Moves polyfill to be a dev dep.

### CODE REVIEW

File: react/package.json [hunk 1]
```diff
@@ -26,18 +26,13 @@
...
-    "@rollup/plugin-terser": "^0.4.3"
+    "typescript": "^5.1.3"
   },
   "peerDependencies": {
     "react": "^18.2.0",
-    "react-dom": "^18.2.0"
-  },
-  "dependencies": {
+    "react-dom": "^18.2.0",
     "dsd-polyfill": "1.0.0"
```
Comment (wkillerud) [inline react/package.json]: Did you want to move `dsd-polyfill` to `devDependencies` here?

Comment (digitalsadhu) [reply react/package.json]: yes. It's only used when building in dev. The final production result is included with the package

Comment (wkillerud) [reply react/package.json]: Sorry, I meant now it's listed as a `peerDependency` and should perhaps be moved to L28,5.

**Review by wkillerud — APPROVED** [review]

File: react/package.json (hunk 1 above)
Comment (digitalsadhu) [reply react/package.json]: Oh! Good catch!

**Pull request merged**